proto/pulumi/language.proto

```diff
@@ -109,4 +115,27 @@ message InstallDependenciesRequest {
 message InstallDependenciesResponse {
     bytes stdout = 1; // a line of stdout text.
     bytes stderr = 2; // a line of stderr text.
+}
+
+message GenerateProjectRequest {
+    string directory = 1;
+    string project = 2;
+    map<string, string> source = 3;
+
+    // Skip generating any project files, just generate the program source.
+    bool sourceOnly = 4;
+    string loaderAddress = 5;
```

Comment:
I have plans here around making schemas available on the registry via URL this would allow codegen (including docs gen!) to work without plugins.

Let's chat this week about how to enable this in the interface. I think `GetSchema` will be deprecated and eventually plugins and schemas are shipped independently as artifacts.